torch/csrc/jit/script/module.h

```diff
@@ -450,6 +427,81 @@ struct TORCH_API Module {
   ModulePtr module_value_;
 };
 
+struct slot_iterator {
+  slot_iterator(Module module, c10::optional<EntityType> type, size_t i)
+      : module_(module), type_(type), i_(i) {
+    advance_to_valid();
+  }
+  Slot operator*() const {
+    return module_._get_slot(i_);
+  }
+  Slot operator->() const {
+    return module_._get_slot(i_);
+  }
+  slot_iterator& operator++() {
+    ++i_;
+    advance_to_valid();
+    return *this;
+  }
+  slot_iterator operator++(int) {
+    slot_iterator old = *this;
+    ++(*this);
+    return old;
+  }
+
+ private:
+  void advance_to_valid() {
+    while (i_ < module_.num_slots() &&
+           (type_ && module_._get_slot(i_).entity_type() != *type_)) {
+      ++i_;
+    }
+  }
+  Module module_;
+  c10::optional<EntityType> type_;
+  size_t i_;
+
+  friend inline bool operator!=(const slot_iterator& a, const slot_iterator& b);
+};
+
+inline bool operator!=(const slot_iterator& a, const slot_iterator& b) {
+  return a.i_ != b.i_;
+}
+
+struct slot_list {
```

Comment:
Can you add a comment explaining what this (and `slot_iterator`) is for/what problem it solves